giskard/scanner/common/loss_based_detector.py

```diff
@@ -95,8 +95,11 @@ def _find_slices(self, model: BaseModel, dataset: Dataset, meta: pd.DataFrame):
         sliced = sf.run(dataset_with_meta, features, target=self.LOSS_COLUMN_NAME)
         slices = sum(sliced.values(), start=[])
 
-        # Keep only slices of size at least 5% of the dataset or 20 samples (whatever is larger)
-        slices = [s for s in slices if max(0.05 * len(dataset), 20) <= len(dataset_with_meta.slice(s))]
+        # Keep only slices of size at least 5% of the dataset or 20 samples (whatever is larger) and conversely exclude
+        # slices which are larger than 95% of the dataset
+        slices = [
+            s for s in slices if max(0.05 * len(dataset), 20) <= len(dataset_with_meta.slice(s)) <= 0.95 * len(dataset)
```

Comment:
What's the harm in detecting issues with slices of > 95%?
wasn't the case of a user who detected a 100% spurious correlation in his model he wasn't aware about, and was super happy about it?

Reply:
- this is just for loss-based detectors, spurious correlation has impact here
- even if it was included, for how spurious correlation is calculated, 100% would just mean that the target had a unique constant value, and it would not make sense to talk about correlation
- for other loss-based detectors, as we filter out slices < 5% of the dataset, it makes sense to also filter out slices which are larger than 95% because they would be compared against a < 5% complementary data slice
- in the end, this filter does not change the results but it allows us to skip a good number of useless comparisons which would have been dropped later anyway

Reply:
Ok clear, Thanks!

Reply:
100% correlation, doesn't necessarily imply a target having a unique constant value no?
It's been a while but wouldn't this:
featureA featureB target
1 1 0
2 1 0
3 0 1
4 0 1
5 1 0
be considered that featureB is 100% correlated with target?